thirdeye/thirdeye-frontend/app/pods/components/self-serve-graph/component.js

```diff
@@ -0,0 +1,80 @@
+/*    {{self-serve-graph
+      isMetricDataLoading=isMetricDataLoading
+      isMetricDataInvalid=isMetricDataInvalid
+      isDimensionFetchDone=isDimensionFetchDone
+      isMetricSelected=isMetricSelected
+      metricData=metricData
+      topDimensions=topDimensions
+      componentId='create-alert'
+    }}*/
+
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+
+export default Component.extend({
+
+isMetricDataLoading: true,
+isMetricDataInvalid: false,
+isDimensionFetchDone: false,
+metricData: {},
+topDimensions: [],
+componentId: '',
+
+graphMessageText: {
+  loading: 'Graph will appear here when data is loaded...',
+  error: 'Could not load graph data for this metric...'
+},
+
+/**
+ * Standard legend settings for graph
+ */
+legendText: {
+  dotted: {
+    text: 'WoW'
+  },
+  solid: {
+    text: 'Observed'
+  }
+},
+
+/**
+ * All selected dimensions to be loaded into graph
+ * @returns {Array}
+ */
+selectedDimensions: computed(
+  'topDimensions',
+  'topDimensions.@each.isSelected',
+  function() {
+    console.log('topDimensions : ', topDimensions);
+    const topDimensions = this.get('topDimensions');
+    return topDimensions ? this.get('topDimensions').filterBy('isSelected') : [];
+  }
+),
+
+didReceiveAttrs() {
```

Comment:
fix: `didReceiveAttrs` is optional if you're not doing anything with it

Reply:
It looks like I missed this file during clean-up.